sdk/trace/sampling.go

```diff
@@ -125,29 +125,101 @@ func NeverSample() Sampler {
 	return alwaysOffSampler{}
 }
 
-// ParentSample returns a Sampler that samples a trace only
-// if the the span has a parent span and it is sampled. If the span has
-// parent span but it is not sampled, neither will this span. If the span
-// does not have a parent the fallback Sampler is used to determine if the
-// span should be sampled.
-func ParentSample(fallback Sampler) Sampler {
-	return parentSampler{fallback}
+// ParentBased returns a composite sampler which behaves differently,
+// based on the parent of the span. If the span has no parent,
+// the root(Sampler) is used to make sampling decision. If the span has
+// a parent, depending on whether the parent is remote and whether it
+// is sampled, one of the following samplers will apply:
+// - remoteParentSampled(Sampler) (default: AlwaysOn)
+// - remoteParentNotSampled(Sampler) (default: AlwaysOff)
+// - localParentSampled(Sampler) (default: AlwaysOn)
+// - localParentNotSampled(Sampler) (default: AlwaysOff)
+func ParentBased(root Sampler, samplers ...SamplerOption) Sampler {
+	return parentBased{
+		root:     root,
+		samplers: configureSamplersForParentBased(samplers),
+	}
+}
+
+type parentBased struct {
+	root     Sampler
+	samplers parentBasedSamplers
+}
+
+type SamplerOption func(*parentBasedSamplers)
+
+type parentBasedSamplers struct {
+	remoteParentSampled, remoteParentNotSampled Sampler
+	localParentSampled, localParentNotSampled   Sampler
+}
+
+func configureSamplersForParentBased(samplers []SamplerOption) parentBasedSamplers {
+	o := parentBasedSamplers{}
+	for _, s := range samplers {
+		s(&o)
+	}
+
+	if o.remoteParentSampled == nil {
+		o.remoteParentSampled = AlwaysSample()
+	}
+	if o.remoteParentNotSampled == nil {
+		o.remoteParentNotSampled = NeverSample()
+	}
+	if o.localParentSampled == nil {
+		o.localParentSampled = AlwaysSample()
+	}
+	if o.localParentNotSampled == nil {
+		o.localParentNotSampled = NeverSample()
+	}
```

Comment:
If you specify the default values in the initial struct declaration then you don't need to check whether they were provided:

```suggestion
	o := parentBasedSamplers{
		remoteParentSampled: AlwaysSample(),
		remoteParentNotSampled: NeverSample(),
		localParentSampled: AlwaysSample(),
		localParentNotSampled: NeverSample(),
	}
	for _, s := range samplers {
		s(&o)
	}
```